### PR TITLE
UniAsserter - introduce getData()/putData() methods

### LINKED ISSUE #29584: UniAsserter - make it easier to share data between assertions
### Description

Currently, if you want to share some data the only way is to use some "external" holder, e.g.
```java
@RunOnVertxContext
public void testSomething(UniAsserter asserter) {
   AtomicReference<Foo> holder = new AtomicReference<>();
   asserter.execute(() -> {
      Foo f = new Foo();
      // do something with Foo
      holder.set(f);
   }
   asserter.assertThat(() -> findAnotherFoo(), f -> assertEquals(f.getName(), holder.get().getName()));
}
```

### Implementation ideas

We could add some method directly to the `UniAsserter` or even consider [Mutiny context passing API](https://smallrye.io/smallrye-mutiny/2.0.0/guides/context-passing/#how-to-access-a-context).

### PR DESCRIPTION
**Pull request by mkouba**
- resolves #29584
- also add some unit tests

### CODE REVIEW

**Comment by mkouba** [comment]
It's not exactly super elegant but it's IMO useful.

File: test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java [hunk 1]
```diff
@@ -49,4 +51,45 @@ public interface UniAsserter {
      * The assertions fails if the Uni does not fail.
      */
     <T> UniAsserter assertFailedWith(Supplier<Uni<T>> uni, Class<? extends Throwable> c);
+    
+    /**
+     * 
+     * @return the test data associated with the {@link #DEFAULT_KEY}, or {@code null} 
+     * @see #getData(String)
+     */
+    default Object getData() {
```
[the remark this answers is not in the document]
Comment (mkouba) [reply test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java:60]: Well, it's convenient if you only work with single object... but I don't insist on it :-)

[the remark this answers is not in the document]
Comment (geoand) [reply test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java:60]: I think it's a little confusing TBH

[the remark this answers is not in the document]
Comment (mkouba) [reply test-framework/vertx/src/main/java/io/quarkus/test/vertx/UniAsserter.java]: Done.

**Review by geoand — APPROVED** [review]
Thanks!

**Pull request merged**